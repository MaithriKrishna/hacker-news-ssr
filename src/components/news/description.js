import React from "react";
import { updateNewsInLocalStorage, getLocalStorageData } from "../helper";

const Hide = ({ news, info, updateModifiedNews }) =>
    <span
        className="highlight clickable"
        onClick={() => {
            updateNewsInLocalStorage(news, info.objectID, true)
            updateModifiedNews(getLocalStorageData(news.page))
        }}>hide</span>;


export const Description = ({ news, info, updateModifiedNews }) =>
    <div className="description">
        <span className="highlight title"><a href={info.url}>{info.title}</a></span>
        <span className="url"> ( <a href={info.url}>{info.url}</a>) by </span>
        <span className="highlight">{info._highlightResult.author.value}</span>
        <span>{info.time} hours ago [</span>
        <Hide news={news} info={info} updateModifiedNews={updateModifiedNews} />
        <span>]</span>
    </div>;