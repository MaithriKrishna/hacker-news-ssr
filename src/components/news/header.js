import React from "react";

export const Header = () =>
    <tr>
        <th>Comments</th>
        <th>Vote Count</th>
        <th>UpVote</th>
        <th>News Details</th>
    </tr>
    ;
