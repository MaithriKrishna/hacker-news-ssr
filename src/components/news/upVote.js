import React from "react";
import { updateNewsInLocalStorage, getLocalStorageData } from "../helper";

export const UpVote = ({ news, info, updateModifiedNews }) =>
    <span
        aria-label="upvote"
        class="upVote clickable"
        onClick={() => {
            updateNewsInLocalStorage(news, info.objectID, false)
            updateModifiedNews(getLocalStorageData(news.page))
        }
        }
    ></span>;
