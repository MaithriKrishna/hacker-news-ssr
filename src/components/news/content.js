import React from "react";
import { UpVote } from "./upVote";
import { Description } from "./description";

export const Content = ({ news, updateModifiedNews }) =>
    news.hits.length && news.hits.map(info =>
        <tr key={info.objectID}>
            <td data-testid="numberof-comments-1">{info.num_comments}</td>
            <td data-testid="up-votes-1">{info.points || 0}</td>
            <td ><UpVote news={news} info={info} updateModifiedNews={updateModifiedNews} /></td>
            <td><Description info={info} news={news} updateModifiedNews={updateModifiedNews} /></td>
        </tr>
    );
