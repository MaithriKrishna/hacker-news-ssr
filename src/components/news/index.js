import React from "react";
import { connect } from 'react-redux';
import { Header } from "./header";
import { Content } from "./content";

const NewsList = ({ news }) => <div>
    <table>
        <Header />
        <Content news={news} />
    </table>
    {console.log(news)}

</div>

const mapStateToProps = (state, props) => {
    const page = (props && props.match.params.page) | 0;
    return {
        news: state.newsList.newsList[page]
    }
}

export const News = connect(mapStateToProps)(NewsList);
