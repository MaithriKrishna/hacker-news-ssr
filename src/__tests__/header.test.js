import React from 'react';
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import { Header } from '../components/news/header';

test('renders header', () => {
  const table = document.createElement('table');
  const tbody = document.createElement('tbody');

    const { container, getByText } = render(<Header />,{
      container: document.body.appendChild(table).appendChild(tbody)
    });
    expect(getByText('Vote Count')).toBeInTheDocument()
    expect(getByText('News Details')).toBeInTheDocument()
    expect(getByText('UpVote')).toBeInTheDocument()
        expect(container).toMatchSnapshot()
})
