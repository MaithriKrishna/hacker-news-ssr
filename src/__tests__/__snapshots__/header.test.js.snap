// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders header 1`] = `
<tbody>
  <tr>
    <th>
      Comments
    </th>
    <th>
      Vote Count
    </th>
    <th>
      UpVote
    </th>
    <th>
      News Details
    </th>
  </tr>
</tbody>
`;
